crates/interpreter/src/gas/calc.rs

```diff
@@ -161,214 +45,13 @@ pub const fn initcode_cost(len: usize) -> u64 {
     cost
 }
 
-/// `SLOAD` opcode cost calculation.
-#[inline]
-pub const fn sload_cost(spec_id: SpecId, is_cold: bool) -> u64 {
-    if spec_id.is_enabled_in(SpecId::BERLIN) {
-        if is_cold {
-            COLD_SLOAD_COST
-        } else {
-            WARM_STORAGE_READ_COST
-        }
-    } else if spec_id.is_enabled_in(SpecId::ISTANBUL) {
-        // EIP-1884: Repricing for trie-size-dependent opcodes
-        ISTANBUL_SLOAD_GAS
-    } else if spec_id.is_enabled_in(SpecId::TANGERINE) {
-        // EIP-150: Gas cost changes for IO-heavy operations
-        200
-    } else {
-        50
-    }
-}
-
-/// Static gas cost for sstore.
-#[inline]
-pub const fn sstore_cost_static(spec_id: SpecId) -> u64 {
-    if spec_id.is_enabled_in(SpecId::BERLIN) {
-        WARM_STORAGE_READ_COST
-    } else if spec_id.is_enabled_in(SpecId::ISTANBUL) {
-        ISTANBUL_SLOAD_GAS
-    } else {
-        SSTORE_RESET
-    }
-}
-
-/// Dynamic gas cost for sstore.
-#[inline]
-pub const fn sstore_cost_dynamic(spec_id: SpecId, vals: &SStoreResult, is_cold: bool) -> u64 {
-    sstore_cost(spec_id, vals, is_cold) - sstore_cost_static(spec_id)
-}
-
-/// Static gas cost for sstore.
-#[inline]
-pub const fn static_sstore_cost(spec_id: SpecId) -> u64 {
-    if spec_id.is_enabled_in(SpecId::BERLIN) {
-        WARM_STORAGE_READ_COST
-    } else if spec_id.is_enabled_in(SpecId::ISTANBUL) {
-        ISTANBUL_SLOAD_GAS
-    } else {
-        SSTORE_RESET
-    }
-}
-
-/// Dynamic gas cost for sstore.
-#[inline]
-pub const fn dyn_sstore_cost(spec_id: SpecId, vals: &SStoreResult, is_cold: bool) -> u64 {
-    sstore_cost(spec_id, vals, is_cold) - static_sstore_cost(spec_id)
-}
-
-/// `SSTORE` opcode cost calculation.
-#[inline]
-pub const fn sstore_cost(spec_id: SpecId, vals: &SStoreResult, is_cold: bool) -> u64 {
-    if spec_id.is_enabled_in(SpecId::BERLIN) {
-        // Berlin specification logic
-        let mut gas_cost = istanbul_sstore_cost::<WARM_STORAGE_READ_COST, WARM_SSTORE_RESET>(vals);
-
-        if is_cold {
-            gas_cost += COLD_SLOAD_COST;
-        }
-        gas_cost
-    } else if spec_id.is_enabled_in(SpecId::ISTANBUL) {
-        // Istanbul logic
-        istanbul_sstore_cost::<ISTANBUL_SLOAD_GAS, SSTORE_RESET>(vals)
-    } else {
-        // Frontier logic
-        frontier_sstore_cost(vals)
-    }
-}
-
-/// EIP-2200: Structured Definitions for Net Gas Metering
-#[inline]
-const fn istanbul_sstore_cost<const SLOAD_GAS: u64, const SSTORE_RESET_GAS: u64>(
-    vals: &SStoreResult,
-) -> u64 {
-    if vals.is_new_eq_present() {
-        SLOAD_GAS
-    } else if vals.is_original_eq_present() && vals.is_original_zero() {
-        SSTORE_SET
-    } else if vals.is_original_eq_present() {
-        SSTORE_RESET_GAS
-    } else {
-        SLOAD_GAS
-    }
```

Comment:
This code is slighly changes as SLOAD_GAS is not part of the static instruction gas. So we have fewer branches

It is easy to follow commnets
https://github.com/bluealloy/revm/blob/403f28f96b569d5d25966d43e9b9d1115965a084/crates/interpreter/src/gas/params.rs#L289-L300